### PR TITLE
Add support for experimental rules, and ER62, ER87

### LINKED ISSUE #845: SIA-R62 should accept difference in `cursor` as distinguishable
Depends on #844

- [ ] Follow up on DEV-12304.

### LINKED ISSUE #952: SIA-R62 should accept more differences in `font`
Currently only accepting differences in `font-weight`.
Seen a case like
```css
p { font-family: VAGRoundedStd-Light}
a { font-family: VAGRoundedStd-Bold}
```

### PR DESCRIPTION
**Pull request by Jym77**
Resolves #952, resolves #845.

* ER87: ask question whose subject is not the test target.
* ER62: send `font` and `cursor` in extended diagnostic, both need UX/frontend work to be displayed correctly.

### CODE REVIEW

File: packages/alfa-rules/src/sia-er62/rule.ts [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+ * in `cursor` (in the `hover` state).
+ * These cannot be easily displayed in the current Page report:
+ * * `cursor` only shows on hovering, we could have a `a:link` and `a:hover`
+ *   states that look similar, but the first one is bad and the second good,
+ *   resulting in confusing info;
+ * * `font-family` may depend on a font that for some reason can't be loaded,
+ *   so the difference wouldn't be visible.
+ */
+export default Rule.Atomic.of<Page, Element>({
+  uri: "https://alfa.siteimprove.com/rules/sia-r62",
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-er62/rule.ts:53]:
```suggestion
  uri: "https://alfa.siteimprove.com/rules/sia-r62#experimental",
```

File: packages/alfa-rules/src/sia-er87/rule.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+ * This version of RD87 ask questions whose subject is not the target of the rule.
+ * The context of the question is still the test target (the document), but the
+ * subjects can be various elements (the first focusable element, or its
+ * destination once it's been identified as a link).
+ * This needs changes in Dory, Nemo, and likely databases to be stored;
+ * this needs changes in the Page Report to be able to highlight an element
+ * different than the test target.
+ */
+export default Rule.Atomic.of<Page, Document, Question, Element>({
+  uri: "https://alfa.siteimprove.com/rules/sia-r87",
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-er87/rule.ts:39]:
```suggestion
  uri: "https://alfa.siteimprove.com/rules/sia-r87#experimental",
```

File: packages/alfa-rules/src/sia-er62/rule.ts [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+    const cursor = context.isHovered(target)
+      ? [["cursor", Serialise.getLonghand(style, "cursor")] as const]
+      : [];
+
+    return ComputedStyles.of(
+      [
+        ...border,
+        ...cursor,
+        ["color", Serialise.getLonghand(style, "color")] as const,
+        ["font", Serialise.font(style)] as const,
```
Comment (zsofitoth) [inline packages/alfa-rules/src/sia-er62/rule.ts:522]: Do the `font` and `cursor` changes that this PR covers make [this PR](https://github.com/Siteimprove/alfa/pull/968/files) obsolete? If yes, should we close that PR when this is merged?

**Review by zsofitoth — COMMENTED** [review]
It looks good to me, just to clarify these rules will move from the experimental state to regular rule once they are no longer breaking for the FE?

**Comment by Jym77** [comment]
> It looks good to me, just to clarify these rules will move from the experimental state to regular rule once they are no longer breaking for the FE?

Yes, the idea that whenever these rules are handled downstream, we move them to regular rule replacing/merging the corresponding regular rule.

**Review by zsofitoth — APPROVED** [review]

**Comment by zsofitoth** [comment]
> > It looks good to me, just to clarify these rules will move from the experimental state to regular rule once they are no longer breaking for the FE?
> 
> Yes, the idea that whenever these rules are handled downstream, we move them to regular rule replacing/merging the corresponding regular rule.

Super then we just have to be extra careful not to replace them or should we add the box shadow changes in this PR as well?

**Comment by Jym77** [comment]
> Super then we just have to be extra careful not to replace them or should we add the box shadow changes in this PR as well?

We'll definitely have to pay the interests on the technical debt whenever we update the "regular" version of the rules 😞 
I think it is better to not constantly try to keep them "synced", so we don't have to constantly think about it. We just know that we will have to be careful when stabilising an experimental rule, but that's part of the process and will only happen once at the end of their lifecycle.

**Pull request merged**